tapestry.js

```diff
@@ -1034,10 +1073,43 @@ function updateViewedValue(id, amountViewedTime, duration) {
     dataset.nodes[index].typeData.progress[0].value = amountViewed;
     dataset.nodes[index].typeData.progress[1].value = amountUnviewed;
 
+    var progressObj = JSON.stringify(getDatasetProgress());
     if (saveProgressToCookie) {
-        var progressObj = JSON.stringify(getDatasetProgress());
-        Cookies.set("progress-data-"+tapestrySlug, progressObj);
-        Cookies.set("h5p-video-settings", h5pVideoSettings);
+        
+        // Save to database if logged in
+        if (tapestryWpUserId) {
+            
+            if (id) {
+                var progData = {
+                    "post_id": tapestryWpPostId,
+                    "node_id": id,
+                    "progress_value": amountViewed
+                };
+                jQuery.post(TAPESTRY_PROGRESS_URL, progData, function(result) {
+                }).fail(function(e) {
+                    console.log("Error with adding progress data");
+                    console.log(e);
+                });
+            }
+
+            if (h5pVideoSettings && !isEmptyObject(h5pVideoSettings)) {
+                var h5pData = {
+                    "post_id": tapestryWpPostId,
+                    "json": JSON.stringify(h5pVideoSettings)
+                };
+                jQuery.post(TAPESTRY_H5P_SETTINGS_URL, h5pData, function(result) {
+                }).fail(function(e) {
```

Comment:
nit: Could you please put the `})` at the end of the previous line? That way, it's more clear that the nested code is the callback for `.fail`.

Reply:
Moved the }) to the previous line